packages/components/src/components/Cell/NavigationCell.tsx

```diff
@@ -29,30 +28,64 @@ export type NavigationCellProps = {
      * Name of the icon to use as a left adornment.
      */
     iconName?: IconName;
+    /**
+     * Additional titles to be shown next to the title section in the cell.
+     */
+    additionalTitlesRight?: string[];
+    /**
+     * Additional titles to be shown under the description section in the cell.
+     */
+    additionalTitlesUnder?: string[];
 } & Omit<CellProps, "leftAdornment" | "rightAdornment" | "onPress">;
-
 export const NavigationCell = ({
     title,
     onPress,
     disabled = false,
     description,
     iconName,
+    additionalTitlesRight,
+    additionalTitlesUnder,
     ...cellProps
 }: NavigationCellProps) => {
     const styles = useStyles(themeStyles);
+    const [maxWidth, setMaxWidth] = useState(0);
 
     const IconAdornment = () => (
         <View style={styles.adornmentContainer}>
             <Icon name={iconName ?? "dots-square"} color={disabled ? "textDisabled" : undefined} />
         </View>
     );
-
     const DisclosureAdornment = () => (
         <View style={styles.adornmentContainer}>
             <Icon name="chevron-right" color={disabled ? "textDisabled" : undefined} />
         </View>
     );
 
+    const renderAdditionalTitles = (titles: string[]) => {
+        return titles?.map((text, index) => (
+            <Typography
+                onLayout={event => {
+                    const width = event.nativeEvent.layout.width;
+                    if (width > maxWidth) {
+                        setMaxWidth(width);
```

Comment:
Setting the state of the component on each render like this has two consequences:
1. The component rerenders a lot of times depending on how many additional titles you have
2. Since you are using the same state variable for both titles under and to the side of the cell, the `maxWidth` variable will represent the largest title of them all. This means that if you have a very short title on the left, but a long one under, the left title will be affected by the titles under.

Instead of using state and layout calculations on the titles, is it possible to instead divide the side titles up into boxes of the same length, and overflow them with a ellipsises (...) if they go beyond this? You could use flex: 1 on each of the titles to ensure they each take up the same amount. Since you want them to align with the left side of the cell, you can use justifyContent: "flex-end" on the parent for example.

Reply:
Can't recall why i did this in the first place. When removing the maxWidth and related logic it seems to be working just fine..